feature/s3/transfermanager/pool.go

```diff
@@ -0,0 +1,252 @@
+
+package transfermanager
+
+import (
+	"context"
+	"fmt"
+	"sync"
+)
+
+type byteSlicePool interface {
+	Get(context.Context) (*[]byte, error)
+	Put(*[]byte)
+	ModifyCapacity(int)
+	SliceSize() int64
+	Close()
+}
+
+type maxSlicePool struct {
```

Comment:
See my comment elsewhere about non-necessity of shared pool. I think we can greatly simplify this by just having one fixed pool per upload call.

Reply:
You removed the shared-ness of the pool but you didn't really simplify anything:
1. The pool sized is now fixed, since it's not shared, it never has to grow or shrink. The entire `ModifyCapacity` API and business logic to support it should not longer be necessary.
2. None of the read/write locking here should be necessary. [Buffered channels](https://go.dev/tour/concurrency/3) inherently give us synchronization across goroutines, this should be as simple as Get() and Put() calls that pull and push to such a channel.

Frankly I would recommend you start from scratch here. A buffered-channel-based fixed size pool should be trivial to implement in significantly less code vs. inheriting/bastardizing the old version of this.